Customrole (#53)
homeaway/datapull: docs/docs/custom_role_creation.md

> **Custom Role Creation for EMR Clusters**

Reviewer: ```suggestion
**Custom EC2 Instance Profile/Role for EMR Clusters**
```

> This is to help the customers who wants to use the custom role than the default `emr_ec2_datapull_role` role. We have created a ![sample terraform file](resources/sample_instanceprofile.tf) for creation of the role.

Reviewer: ```suggestion
By default, DataPull on EMR uses the IAM role `emr_ec2_datapull_role` as the [service role for cluster EC2 instances](https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-iam-role-for-ec2.html) (also called the EC2 instance profile for Amazon EMR).  This  IAM role `emr_ec2_datapull_role` has sufficient access to run DataPull jobs that involve non-AWS data stores (for example, you can move data from a Cassandra cluster to a MongoDB cluster without having to read this document). However, if you want to have DataPull read/write data from an AWS data store like S3, the recommendation is that you create a custom EC2 Instance Profile/Role and grant that tole access to the AWS datastore/resource that DataPull needs to access.  Here is a
[sample terraform file](resources/sample_instanceprofile.tf) for the creation of a custom IAM role that can be used by DataPull on EMR as a custom EC2 Instance Profile/Role. The IAM Role thus used, can be specified for each DataPull job as part of the input JSON.
```

> We have two policies, one for granting required permissions for accessing the s3 bucket and the other is for retrieving secrets from AWS secrets manager.

Reviewer: ```suggestion
The [sample terraform file](resources/sample_instanceprofile.tf) includes some required policies; and also some optional policies that allow DAtaPUll to interact with AWS Secrets Manager, your own S3 bucket, etc. Please remove the optional policies if you do not need them. 
```

> We can remove any of the policies if not needed along with the policy attachment statements.  

Reviewer: ```suggestion
Please remove the optional policies and their related policy attachments, if you do not need them. 
```

> We should add the  name of the policy to `datapull_passrole_policy` under the `Resource` section of the policy. such that the api will have access to spin up the EC2 instances with the custom role. 

Reviewer: Please consider rewriting this
